File: app/src/main/java/Entity/Order.java
package Entity;

import org.json.JSONObject;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;

@Data
@Builder
@Getter
public class Order {
    private int id;
    private String imageURL;
    private int status;
    private double total;
    private int quantity;
    private String reason;
    private String createAt;
    private String description;

    public static Order generateOrderFromJSON(JSONObject jsonObject) {
        try {
            return Order.builder()
                    .id(jsonObject.getInt("id"))
                    .imageURL(jsonObject.getJSONObject("Image_model").getString("imageURL"))
                    .status(jsonObject.getInt("status"))
                    .quantity(jsonObject.getInt("quantity"))
                    .total(jsonObject.getDouble("total"))
                    .reason(jsonObject.getString("reason"))
                    .createAt(jsonObject.getString("createAt"))
                    .description(jsonObject.getString("description"))
                    .build();
        } catch (Exception e) {
            return null;
        }
    }
}

File: Nodejs/routings/Order.js
var express = require('express');
var router = express.Router();
const { checkToken } = require('../helpers/TokenCheck')
const { sequelize } = require('../databases/database')
const OrderModel = require('../models/Order')(sequelize)
const OrderDetail = require('../models/OrderDetail')(sequelize)
const ImageModel = require('../models/Image')(sequelize)
const UserModel = require('../models/User')(sequelize)
const CommentStoreModel = require('../models/StoreComment')(sequelize)
const { validateString } = require('../validations/validate')
const { validationResult } = require('express-validator')
const { Op, INTEGER } = require("sequelize");


OrderModel.hasMany(OrderDetail, { foreignKey: 'orderId' })
OrderDetail.belongsTo(OrderModel, { foreignKey: 'orderId' })

ImageModel.hasMany(OrderModel, { foreignKey: 'imageId' })
OrderModel.belongsTo(ImageModel, { foreignKey: 'imageId' })

/**
 * URL: http://localhost:3000/order/getall
 */
router.post('/getall', validateString(), async(req, res) => {
        //validate du lieu tu client gui len
        const { tokenkey } = req.headers
        const isValidToken = await checkToken({ tokenkey })
        if (isValidToken == false) {
            res.json({
                result: "TK01"
            })
            return;
        }
        const { userId } = req.body
        try {

            let foundStore = await OrderModel.findAll({
                where: {
                    userId: {
                        [Op.eq]: userId
                    }
                },
                include: [{
                    model: ImageModel,
                    as: "Image_model",
                    attributes: ['imageURL']
                }]
            })

            res.json({
                result: 'SC',
                data: foundStore
            })
        } catch (exception) {
            res.status(500).json({
                result: 'E500',
                message: `Error details: ${exception.toString()}`
            })
        }
    })
    /**
     * URL: http://localhost:3000/foods/addnew
     */
router.post('/add', async(req, res) => {
    //validate du lieu tu client gui len    
    const { userId, quantity, total, imageId, description } = req.body
    try {
        var status = "1";
        const expireDate = new Date();
        let newFood = await OrderModel.create({
            userId,
            imageId,
            quantity,
            total,
            description,
            status,
            expireDate
        })
        await newFood.save()

        res.json({
            result: 'SC',
            data: newFood
        })
    } catch (exception) {
        res.status(500).json({
            result: 'E500',
            message: `Error details: ${exception.toString()}`
        })
    }
})

module.exports = router